### PR TITLE
Remove prng maxsize restriction

### PR DESCRIPTION
**Pull request by mattjj**

### CODE REVIEW

File: jax/random.py [hunk 1]
```diff
@@ -196,23 +197,45 @@ def _fold_in(key, data):
...
+    return bits
+
+  if num_chunks == 1:
+    out = randbits_onekey(key, n)
+  else:
+    dtype = onp.uint64 if bit_width == 64 else onp.uint32
+    out = lax.full((prod(shape),), lax.tie_in(key, dtype(0)), dtype)
+    for k in split(key, num_chunks):
+      bits = randbits_onekey(k, min(n, _max_randvals_per_key))
+      out = lax.dynamic_update_slice(out, bits, onp.array([out.shape[0] - n]))
```
Comment (hawkinsp) [inline jax/random.py]: I would have thought that concatenating them was more natural.

File: jax/random.py [hunk 2]
```diff
@@ -196,23 +197,45 @@ def _fold_in(key, data):
...
-
-  counts = lax.tie_in(key, lax.iota(onp.uint32, max_count))
-  bits = threefry_2x32(key, counts)
-  if bit_width == 64:
-    bits = [lax.convert_element_type(x, onp.uint64) for x in np.split(bits, 2)]
-    bits = (bits[0] << onp.uint64(32)) | bits[1]
-  return lax.reshape(bits, shape)
+
+  n = (bit_width // 32) * prod(shape)  # number of 32-bit values to generate
+  num_chunks, extra = divmod(n, _max_randvals_per_key)
```
Comment (hawkinsp) [inline jax/random.py]: Perhaps define a helper function `ceil_of_ratio` in the way that XLA does in C++? Would make it clearer.

File: jax/random.py (hunk 1 above)
Comment (mattjj) [reply jax/random.py]: It was, and that's how I wrote it at first, but then I decided I might roll this into a fori_loop. Then I got lazy... I can put it back to a concatenate.

**Comment by hawkinsp** [comment]
This was subsequently fixed!

**Pull request closed without merging**